### PR TITLE
ZEN-27507: delete relationship between products and product classes t…

### PR DESCRIPTION
**Pull request by fraibacas**
…o reduce ConflictErrors

### CODE REVIEW

File: Products/ZenModel/ProductClass.py [hunk 1]
```diff
@@ -108,18 +110,27 @@ def __init__(self, id, title="", prodName=None,
...
+        if count:
+            return model_catalog.search(query=query, limit=0)
+        else:
+            return model_catalog.search(query=query)
 
-    def count(self):
-        """Return the number of existing instances for this class.
-        """
-        return self.instances.countObjects()
+    def instances(self):
```
Comment (iancmcc) [inline Products/ZenModel/ProductClass.py]: This is too expensive. Make it a generator.

```
for brain in self._find_instances_in_catalog():
    if brain:
        yield brain.getObject()
```

File: Products/Zuul/catalog/model_catalog.py [hunk 2]
```diff
@@ -131,7 +131,7 @@ def model_index(self):
         return self._data_manager.model_index
 
     def _get_forbidden_classes(self):
-        return (Software, OperatingSystem)
+        return ()
```
Comment (iancmcc) [inline Products/Zuul/catalog/model_catalog.py]: Why?

File: Products/ZenModel/ProductClass.py [hunk 3]
```diff
@@ -52,8 +54,9 @@ class ProductClass(ZenModelRM, ZenPackable):
     )
 
     _relations = ZenPackable._relations + (
-        ("instances", ToMany(ToOne, "Products.ZenModel.MEProduct", "productClass")),
         ("manufacturer", ToOne(ToManyCont,"Products.ZenModel.Manufacturer","products")),
+        # Deprecated
```
Comment (iancmcc) [inline Products/ZenModel/ProductClass.py]: The migrate script drops the relationship, so you can just drop this. It isn't deprecated so much as gone.

File: Products/Zuul/catalog/model_catalog.py (hunk 2 above)
Comment (fraibacas) [inline Products/Zuul/catalog/model_catalog.py]: Software and Operating systems are MEProduct. If we don't index them we will not be able to get the instances for the related ProductClass

**Pull request merged**